### DISCUSSION TITLE
What does `poetry show` actually tell me?

### DISCUSSION THREAD

**Discussion by elliot-100** [python-poetry/poetry#7311] [Q&A] [answered]
Hi, https://python-poetry.org/docs/cli/#show just says 

> "To list all the **available packages**, you can use the show command."

But this seems vague.
It can't just be those that are installed, as I know package names are shown in red (or annotated with `(!)`) if they're not installed.

The code refers to locked packages:
https://github.com/python-poetry/poetry/blob/81c7d00cb3985c6db81d8370ef400731b05bf047/src/poetry/console/commands/show.py#L295

So, should the doc read something like:

> "List all packages specified in the lock file, with their versions. Packages not installed appear in red or or annotated with `(!)`.

?

(As a side note, `poetry show <package-name>` doesn't tell you anything about installation status)

Happy to contribute to docs if someone can confirm my understanding. Thanks!

**Comment by radoering** [comment, marked as answer]
Yeah, PRs improving the documentation are welcome. See also https://github.com/python-poetry/poetry/discussions/6330#discussioncomment-3527109 for more details about the colors.
